packages/core/documentation/src/components/ProductDocumentModal.vue

```diff
@@ -0,0 +1,398 @@
+          appearance="select"
+          class="document-parent-select"
+          data-testid="documentation-parent"
+          enable-filtering
+          :items="availableParentDocuments"
+          :label="i18n.t('documentation.form_modal.parent_document_select_label')"
+          width="100%"
+        />
+      </template>
+
+      <template #action-buttons>
+        <div class="action-buttons">
+          <KButton
+            v-if="isEditing"
+            appearance="danger"
+            class="edit-documentation-delete-button"
+            data-testid="edit-documentation-delete-button"
+            :disabled="state.matches('pending')"
+            :icon="state.matches('pending') ? 'spinner' : undefined"
+            @click="emit('delete')"
+          >
+            {{ i18n.t('documentation.form_modal.delete_button_text') }}
+          </KButton>
+
+          <div class="button-spacing">
+            <KButton
+              appearance="outline"
+              class="edit-documentation-cancel-button"
+              data-testid="edit-documentation-cancel-button"
+              @click="handleClickCancel"
+            >
+              {{ i18n.t('documentation.form_modal.cancel_button_text') }}
+            </KButton>
+            <KButton
+              appearance="primary"
+              data-testid="edit-documentation-save-button"
+              :disabled="state.matches('pending') || saveDisabled"
+              :icon="state.matches('pending') ? 'spinner' : undefined"
+              @click="handleClickSave"
+            >
+              {{ i18n.t('documentation.form_modal.save_button_text') }}
+            </KButton>
+          </div>
+        </div>
+      </template>
+    </KPrompt>
+  </Teleport>
+</template>
+
+<script setup lang="ts">
+import { computed, onMounted, ref, reactive } from 'vue'
+import { createMachine } from 'xstate'
+import { useMachine } from '@xstate/vue'
+import composables from '../composables'
+import { cloneDeep } from '../helpers'
+import externalLinks from '../external-links'
+
+import type { PropType } from 'vue'
+import type { DocumentListItem, FormData } from 'src/types'
+
+const props = defineProps({
+  documents: {
+    type: Array as PropType<DocumentListItem[]>,
+    required: true,
+  },
+  errorMessage: {
+    type: String,
+    default: '',
+  },
+  hidePublishToggle: {
+    type: Boolean,
+    default: false,
+  },
+  isEditing: {
+    type: Boolean,
+    default: false,
+  },
+  record: {
+    type: Object,
+    default: null,
+  },
+})
+
+const emit = defineEmits<{
+  (e: 'canceled'): void,
+  (e: 'save', formData: FormData, selectedFile: any): void,
+  (e: 'delete'): void,
+}>()
+
+const { i18n } = composables.useI18n()
+const filePlaceholderText = computed(() => props.record.file?.filename)
+const namePlaceholderText = computed(() => selectedFile.value ? selectedFile.value.name?.split('.')[0] : '')
+const publishModel = ref<boolean>(false)
+
+const { state, send } = useMachine(createMachine({
```

Comment:
issue: why are we using XState here? That creates a large dependency and isn’t really needed here is it?

Reply:
removed